### PR TITLE
Improve blend shader and fix text shadow reftest bug.

### PR DESCRIPTION
**Pull request by glennw**
Rewrite the blur shader - it's still not *exactly* the same as gecko, but it's *much* closer.

The previous blur shader was written before we switched image inputs to use pre-multiplied alpha, so was giving incorrect results. At the same time, optimize the blur significantly by making use of the incremental gaussian technique from GPU Gems 3.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1716)

### CODE REVIEW

**Comment by glennw** [comment]
r? @kvark 

cc @Gankro This fixes the text shadow blending bug. In the reftest, it's a fuzzy result - there are a small number of pixels that end up 1.0 / 255.0 from the expected result.

**Comment by glennw** [comment]
Fixes #1710

**Comment by glennw** [comment]
Gecko try (pending):

https://hg.mozilla.org/try/rev/24587a7823fbf5662cbed68176cd8d5dc61f1cb0
https://treeherder.mozilla.org/#/jobs?repo=try&revision=24587a7823fbf5662cbed68176cd8d5dc61f1cb0

File: webrender/res/cs_blur.glsl [hunk 1]
```diff
@@ -65,41 +65,41 @@ void main(void) {
...
 
-        vec2 st0 = clamp(vUv.xy + offset, vUvRect.xy, vUvRect.zw);
-        vec4 color0 = texture(sCacheRGBA8, vec3(st0, vUv.z));
+    for (int i=1 ; i <= vBlurRadius/2 ; ++i) {
+        vec2 offset = vOffsetScale * float(i);
 
-        vec2 st1 = clamp(vUv.xy - offset, vUvRect.xy, vUvRect.zw);
-        vec4 color1 = texture(sCacheRGBA8, vec3(st1, vUv.z));
+        vec2 st0 = clamp(vUv.xy - offset, vUvRect.xy, vUvRect.zw);
+        avg_color += texture(sCacheRGBA8, vec3(st0, vUv.z)) * gauss_coefficient.x;
```
Comment (kvark) [inline webrender/res/cs_blur.glsl]: nit: lets use `textureLod`

**Comment by kvark** [comment]
Push try looks good, merging.
@bors-servo r+

**Pull request merged**